native/native_main.py

```diff
@@ -153,7 +158,12 @@ def handleMessage(message):
             file.write(message["content"])
 
     elif cmd == 'temp':
-        (handle, filepath) = tempfile.mkstemp()
+        prefix = message.get('hostname')
+        if prefix is not None:
+            prefix = sanitizePrefix(prefix)
```

Comment:
This should be in src/native_backgrounds.ts I think.

It is probably a good idea to sanitise things. A quick search found this which looks sensible: https://stackoverflow.com/questions/295135/turn-a-string-into-a-valid-filename?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa#answer-295466